internal/venice-common/src/main/java/com/linkedin/venice/meta/ReadOnlyStoreView.java

```diff
@@ -0,0 +1,143 @@
+package com.linkedin.venice.meta;
+
+import com.linkedin.venice.exceptions.StoreVersionNotFoundException;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.views.MaterializedView;
+import com.linkedin.venice.views.VeniceView;
+import com.linkedin.venice.views.ViewUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+
+public class ReadOnlyStoreView extends ReadOnlyStore {
+  private final Map<Integer, ReadOnlyVersion> viewVersionMap;
+
+  public ReadOnlyStoreView(Store delegate, Map<Integer, ReadOnlyVersion> viewVersionMap) {
```

Comment:
I think we need to pass the `viewName` here, so that we can override `getName` and `getStoreName` in `ReadOnlyMaterializedViewVersion`, right?

Reply:
One reason we need this IMO is that we would like to treat each view as an individual store from the interface POV.

Reply:
Yeah, we will need to do this if we want to also provide capabilities to register listeners for view stores and trigger change events for all view stores when the original store changes.